test/gtest/uct/test_amo_fand_for.cc

```diff
@@ -0,0 +1,145 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2018.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "test_amo.h"
+
+
+class uct_amo_fand_for_test : public uct_amo_test {
+public:
+
+    ucs_status_t fand32(uct_ep_h ep, worker& worker, const mapped_buffer& recvbuf,
+                        uint64_t *result, completion *comp) {
+        comp->self     = this;
+        comp->uct.func = atomic_reply_cb;
+        return uct_ep_atomic32_fetch_nb(ep, UCT_ATOMIC_OP_AND, (uint32_t)worker.value,
+                                        (uint32_t*)result, recvbuf.addr(), recvbuf.rkey(),
+                                        &comp->uct);
+    }
+
+    ucs_status_t fand64(uct_ep_h ep, worker& worker, const mapped_buffer& recvbuf,
+                        uint64_t *result, completion *comp) {
+        comp->self     = this;
+        comp->uct.func = atomic_reply_cb;
+        return uct_ep_atomic64_fetch_nb(ep, UCT_ATOMIC_OP_AND, worker.value,
+                                        (uint64_t*)result, recvbuf.addr(), recvbuf.rkey(),
+                                        &comp->uct);
+    }
+
+    ucs_status_t for32(uct_ep_h ep, worker& worker, const mapped_buffer& recvbuf,
+                       uint64_t *result, completion *comp) {
+        comp->self     = this;
+        comp->uct.func = atomic_reply_cb;
+        return uct_ep_atomic32_fetch_nb(ep, UCT_ATOMIC_OP_OR, (uint32_t)worker.value,
+                                        (uint32_t*)result, recvbuf.addr(), recvbuf.rkey(),
+                                        &comp->uct);
+    }
+
+    ucs_status_t for64(uct_ep_h ep, worker& worker, const mapped_buffer& recvbuf,
```

Comment:
maybe make these functions templated on the operation code? to avoid copy/paste
```
template <uct_atomic_op_t op>
ucs_status_t fop32(..)
```

Reply:
good idea... all such op-calls are unified (I left original add/swap calls to provide backward compatibility, will remove it on deprecated code cleaning)